### ISSUE TITLE
RealmResults.toFlow() extension does disk I/O

### ISSUE THREAD

**Issue by BoD** [realm/realm-java#7500]
#### Goal
We are migrating our project to using Flow in our codebase (we used to use LiveData).

Thus we call `toFlow` on our `RealmResults`.

#### Actual Results
When doing so, we notice StrictMode complaining about this method reading from the disk - and since we are calling this from our ViewModel's init block (which I think is fairly reasonable), we are in the main thread at that point.

Here's the StrictMode stacktrace:

```
2021-06-15 11:51:25.085 12924-12924/eu.qonto.qonto.debug D/StrictMode: StrictMode policy violation; ~duration=19 ms: android.os.strictmode.DiskReadViolation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1596)
        at libcore.io.BlockGuardOs.access(BlockGuardOs.java:71)
        at libcore.io.ForwardingOs.access(ForwardingOs.java:72)
        at android.app.ActivityThread$AndroidOs.access(ActivityThread.java:7533)
        at java.io.UnixFileSystem.checkAccess(UnixFileSystem.java:281)
        at java.io.File.exists(File.java:815)
        at io.realm.RealmConfiguration.realmExists(RealmConfiguration.java:257)
        at io.realm.RealmCache.doCreateRealmOrGetFromCache(RealmCache.java:423)
        at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:412)
        at io.realm.Realm.getInstance(Realm.java:424)
        at io.realm.internal.coroutines.InternalFlowFactory$from$3.invokeSuspend(InternalFlowFactory.kt:122)
        at io.realm.internal.coroutines.InternalFlowFactory$from$3.invoke(Unknown Source:10)
        at kotlinx.coroutines.flow.ChannelFlowBuilder.collectTo$suspendImpl(Builders.kt:341)
        at kotlinx.coroutines.flow.ChannelFlowBuilder.collectTo(Unknown Source:0)
        at kotlinx.coroutines.flow.CallbackFlowBuilder.collectTo(Builders.kt:355)
        at kotlinx.coroutines.flow.internal.ChannelFlow$collectToFun$1.invokeSuspend(ChannelFlow.kt:60)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.EventLoop.processUnconfinedEvent(EventLoop.common.kt:69)
        at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:385)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
        at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
        at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
        at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
        at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
        at androidx.lifecycle.BlockRunner.maybeRun(CoroutineLiveData.kt:174)
        at androidx.lifecycle.CoroutineLiveData.onActive(CoroutineLiveData.kt:240)
        at androidx.lifecycle.LiveData.changeActiveCounter(LiveData.java:390)
        at androidx.lifecycle.LiveData$ObserverWrapper.activeStateChanged(LiveData.java:466)
        at androidx.lifecycle.LiveData$LifecycleBoundObserver.onStateChanged(LiveData.java:425)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:354)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:265)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:307)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:148)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:134)
        at androidx.lifecycle.ReportFragment.dispatch(ReportFragment.java:68)
        at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostStarted(ReportFragment.java:187)
        at android.app.Activity.dispatchActivityPostStarted(Activity.java:1362)
        at android.app.Activity.performStart(Activity.java:8061)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3475)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

When digging a bit, we can see that this is provoked by calling `Realm.getInstance(config)` line 122 of `InternalFlowFactory`.

#### Steps & Code to Reproduce
Call `toFlow` from the main thread, with StrictMode enabled.

#### Version of Realm and tooling

Realm version(s): 10.5.0

Realm Sync feature enabled: No

Android Studio version: 4.2.1

Android Build Tools version: not sure / the latest

Gradle version: 7.0.2

Which Android version and device(s): sdk 30 (emulator).

**Comment by cmelchior** [comment]
It seems to be complaining about this line: https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/java/io/realm/RealmConfiguration.java#L257

Which is checking that the Realm file actually exists, and is part of our current caching logic. It seems a bit extreme to complain about disk I/O in that case 🤔, but we could probably refactor the logic as the check is only used within another `if`-statement that doesn't use the disk.

**Comment by dwechsler34lab** [comment]
bitcoin:bc1qxtzauq70pwad35ly4n5c47f7wew9f5gu39mzds?amount=5&label=ZETTAS%20GOLDEN%20CHILD

**Comment by dwechsler34lab** [comment]
I dnt kno y my stuf keps getn erased help plz

**Comment by bmunkholm** [comment]
@dwechsler34lab I don't know what you mean. What is getting deleted?

**Comment by cmelchior** [comment]
A fix has been merged and is part of the next release (10.6.1)
